common/workunit/workunit.cpp

```diff
@@ -2664,41 +2664,19 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
         Owned<CLocalWorkUnit> cw = new CDaliWorkUnit(conn, (ISecManager *) NULL, NULL);
         return &cw->lockRemote(false);
     }
-
     virtual IConstWorkUnitIterator* getWorkUnitsByOwner(const char * owner, ISecManager *secmgr, ISecUser *secuser)
     {
         StringBuffer path("*");
         if (owner && *owner)
-            path.append("[@submitID=\"").append(owner).append("\"]");
-        return _getWorkUnitsByXPath(path.str(), secmgr, secuser);
-    }
-    IConstWorkUnitIterator* getWorkUnitsByState(WUState state, ISecManager *secmgr, ISecUser *secuser)
-    {
-        StringBuffer path("*");
-        path.append("[@state=\"").append(getEnumText(state, states)).append("\"]");
-        return _getWorkUnitsByXPath(path.str(), secmgr, secuser);
-    }
-    IConstWorkUnitIterator* getWorkUnitsByECL(const char* ecl, ISecManager *secmgr, ISecUser *secuser)
-    {
-        StringBuffer path("*");
-        if (ecl && *ecl)
-            path.append("[Query/Text=~\"*").append(ecl).append("*\"]");
+            path.append("[@submitID=?~\"").append(owner).append("\"]");
         return _getWorkUnitsByXPath(path.str(), secmgr, secuser);
     }
-    IConstWorkUnitIterator* getWorkUnitsByCluster(const char* cluster, ISecManager *secmgr, ISecUser *secuser)
+    IConstWorkUnitIterator* getScheduledWorkUnits(ISecManager *secmgr, ISecUser *secuser)
```

Comment:
filtering by state is going to be a common one, even if done via generic mechanism now, I'd have thought.
Wouldn't keeping getWorkUnitsByState vs specific getScheduledWorkUnits be worthwhile during this slim down?

Although, I can imagine getScheduledWorkUnits might morph into an implementation which isn't scanning workunits at all..